File: packages/domain/src/adapters/federal/shared/kill-switch.ts
/**
 * Federal adapter kill-switch identifiers (BB-046 / BB-035 naming).
 * Stable logical ids use the `adapter:<adapterId>` prefix even when the ops
 * kill-switch package is not yet merged.
 */

/** BB-035 logical kill-switch prefix for source adapters. */
export const FEDERAL_ADAPTER_KILL_SWITCH_PREFIX = 'adapter:' as const;

export function federalAdapterKillSwitchId(adapterId: string): string {
  const trimmed = adapterId.trim();
  if (!trimmed) {
    throw new Error('adapterId is required for kill-switch id');
  }
  return `${FEDERAL_ADAPTER_KILL_SWITCH_PREFIX}${trimmed}`;
}

export function parseFederalAdapterKillSwitchId(killSwitchId: string): string | null {
  if (!killSwitchId.startsWith(FEDERAL_ADAPTER_KILL_SWITCH_PREFIX)) {
    return null;
  }
  const adapterId = killSwitchId.slice(FEDERAL_ADAPTER_KILL_SWITCH_PREFIX.length).trim();
  return adapterId || null;
}

File: workers/research/src/black_book_research/adapters/federal/shared/kill_switch.py
"""Federal adapter kill-switch identifiers (BB-046 / BB-035 naming)."""

from __future__ import annotations

FEDERAL_ADAPTER_KILL_SWITCH_PREFIX = "adapter:"


def federal_adapter_kill_switch_id(adapter_id: str) -> str:
    trimmed = adapter_id.strip()
    if not trimmed:
        raise ValueError("adapter_id is required for kill-switch id")
    return f"{FEDERAL_ADAPTER_KILL_SWITCH_PREFIX}{trimmed}"


def parse_federal_adapter_kill_switch_id(kill_switch_id: str) -> str | None:
    if not kill_switch_id.startswith(FEDERAL_ADAPTER_KILL_SWITCH_PREFIX):
        return None
    adapter_id = kill_switch_id[len(FEDERAL_ADAPTER_KILL_SWITCH_PREFIX) :].strip()
    return adapter_id or None
